src/main/java/io/supertokens/webserver/RequestStats.java

```diff
@@ -0,0 +1,130 @@
+
+    // The 2 arrays below contains stats for a day for every minute
+    // the array is stored in such a way that array[currentMinute % MAX_MINUTES] contains the stats for a day ago
+    // until array[(currentMinute - 1) % MAX_MINUTES] which contains the stats for the last minute, circling around
+    // from end of array to the beginning
+    // for e.g. if currentMinute % MAX_MINUTES = 250,
+    // then array[250] contains stats for now - 1440 minutes
+    // array[251] contains stats for now - 1439 minutes
+    // ...
+    // array[1439] contains stats for now - 1191 minutes
+    // array[0] contains stats for now - 1190 minutes
+    // array[1] contains stats for now - 1189 minutes
+    // ...
+    // array[249] contains stats for now - 1 minute
+    private final double[] averageRequestsPerSecond;
+    private final int[] peakRequestsPerSecond;
+
+    private RequestStats() {
+        currentMinute = System.currentTimeMillis() / 60000;
+        currentMinuteRequestCounts = new int[60];
+
+        averageRequestsPerSecond = new double[MAX_MINUTES];
+        peakRequestsPerSecond = new int[MAX_MINUTES];
+        for (int i = 0; i < MAX_MINUTES; i++) {
+            averageRequestsPerSecond[i] = -1;
+            peakRequestsPerSecond[i] = -1;
+        }
+    }
+
+    private void checkAndUpdateMinute(long currentSecond) {
+        if (currentSecond / 60 == currentMinute) {
+            return; // stats update not required
+        }
+
+        int sum = 0;
+        int max = 0;
+        for (int i = 0; i < 60; i++) {
+            sum += currentMinuteRequestCounts[i];
+            max = Math.max(max, currentMinuteRequestCounts[i]);
+        }
+
+        averageRequestsPerSecond[(int) (currentMinute % MAX_MINUTES)] = sum / 60.0;
+        peakRequestsPerSecond[(int) (currentMinute % MAX_MINUTES)] = max;
+
+        // fill zeros for passed minutes
+        for (long i = currentMinute + 1; i < currentSecond / 60; i++) {
+            averageRequestsPerSecond[(int) (i % MAX_MINUTES)] = 0;
+            peakRequestsPerSecond[(int) (i % MAX_MINUTES)] = 0;
+        }
+
+        currentMinute = currentSecond / 60;
+        for (int i = 0; i < 60; i++) {
+            currentMinuteRequestCounts[i] = 0;
+        }
+    }
+
+    private void updateCounts(long currentSecond) {
+        currentMinuteRequestCounts[(int) (currentSecond % 60)]++;
+    }
+
+    public static RequestStats getInstance(Main main, AppIdentifier appIdentifier) {
+        try {
+            return (RequestStats) main.getResourceDistributor()
+                    .getResource(appIdentifier.getAsPublicTenantIdentifier(), RESOURCE_KEY);
+        } catch (TenantOrAppNotFoundException e) {
+            return (RequestStats) main.getResourceDistributor()
+                    .setResource(appIdentifier.getAsPublicTenantIdentifier(), RESOURCE_KEY, new RequestStats());
+        }
+    }
+
+    synchronized public void updateRequestStats(boolean updateCounts) {
```

Comment:
create a cversion of this wirhour boolean that is exposed as a public func

Reply:
Done